cymetric/metrics.py

```diff
@@ -282,16 +282,15 @@ def agents(entry, exit, decom, info):
     significant amounts of missing data.
     """
     mergeon = ['SimId', 'AgentId']
-    ent = tools.raw_to_series(entry, ['SimId', 'AgentId'], 'Kind')
-    idx = ent.index
+    idx = ['SimId','AgentId','ExitTime']
     df = entry[['SimId', 'AgentId', 'Kind', 'Spec', 'Prototype', 'ParentId',
                 'Lifetime', 'EnterTime']]
-    if exit is None:
-        agent_exit = pd.Series(index=idx, data=[np.nan] * len(idx))
-        agent_exit.name = 'ExitTime'
-    else:
-        agent_exit = agent_exit.reindex(index=idx)
-    df = pd.merge(df, agent_exit.reset_index(), on=mergeon)
+    df['ExitTime'] = [np.nan]*len(entry)
+    if exit is not None:
+        exit.columns = ['SimId','AgentId','Exits']
+        df = tools.merge_and_fillna_col(df, exit[['SimId','AgentId','Exits']], 
+                                        'ExitTime','Exits',on=mergeon)
```

Comment:
pep8 (our formatting guidelines) recommends space after commas, and there is trailing space at the end of the line.

Reply:
* pep8 is not our guideline but the one we try to follow :)